system/logaccess/ElasticStack/ElasticStackLogAccess.cpp

```diff
@@ -262,6 +266,46 @@ void processESJsonResp(const cpr::Response & retrievedDocument, StringBuffer & r
     }
 }
 
+/*
+ * Transform ES query response to back-end agnostic response
+ *
+ */
+void processESSearchJsonResp(const cpr::Response & retrievedDocument, StringBuffer & returnbuf, LogAccessLogFormat format)
+{
+    if (retrievedDocument.status_code != 200)
+        throw makeStringExceptionV(-1, "ElasticSearch request failed: %s", retrievedDocument.text.c_str());
+
+#ifdef _DEBUG
+    DBGLOG("Retrieved ES JSON DOC: %s", retrievedDocument.text.c_str());
+#endif
+
+    Owned<IPropertyTree> tree = createPTreeFromJSONString(retrievedDocument.text.c_str());
+    if (!tree)
+        throw makeStringExceptionV(-1, "%s: Could not parse ElasticSearch query response", COMPONENT_NAME);
+
+    if (tree->getPropBool("timed_out", 0))
+        LOG(MCuserProgress,"ES Log Access: timeout reported");
+    if (tree->getPropInt("_shards/failed",0) > 0)
+        LOG(MCuserProgress,"ES Log Access: failed _shards reported");
+
+    DBGLOG("ES Log Access: hit count: '%d'", tree->getPropInt("hits/total/value"));
+
+    processHitsJsonResp(tree->getElements("hits/hits/fields"), returnbuf, format);
```

Comment:
result of getElements will leak.  Worth running this code under valgrind (there is an option in the helm chart to enable that).

Reply:
This sounds like a great feature the entire organization should exploit.
Even though it seems straightforward and simple to use, I did not see any documentation and the helm template wasn't obvious enough for me to utilize it.
It seemed roxie centric by the way. Anyway, can you provide "getting started" type info?
Perhaps startall.sh should provide an explicit valgrind option.